plugins/search/src/components/SearchModal/useSearchModal.tsx

```diff
@@ -120,7 +120,7 @@ export function useSearchModal(initialState = false) {
   const toggleModal = useCallback(
     () =>
       setState(prevState => ({
-        open: true,
+        open: !prevState.open,
```

Comment:
To give context, the `true` here is intentional, as explained in this pull request: https://github.com/backstage/backstage/pull/9364.

Reply:
Hmm, okay.  But I'm curious why the `hidden` doesn't work as intended then. When I click "outside" of the search modal, it doesn't actually hide. Just the "overlay" gets hidden.

Reply:
Let me try following your recommendation and see if that works.

Reply:
Looking at the Dialog API, I don't see any support for a `hidden` property.  It seems `open` controls whether the modal renders or not and if the intention is to keep the model `open` as described in #9364 then doesn't that mean we can't close the modal when `open` is `true`?

Reply:
I couldn't simulate the error in the master. Here is how we are closing the modal in the example app, could you provide us with how you are rendering your search modal?

Reply:
To ensure I understand correctly, the issue mentioned in the pull request description was about the clear button. But you also found that clicking on the modal's background is not closing the modal for you, and you wanted to fix it, or are you trying to add a close button on the modal and manually close it when the button is clicked?

In case you are trying to close the modal manually, here is an example:
https://github.com/backstage/backstage/blob/0fcedaa881f6e8713c6862c78ce464f4420abc31/plugins/search/src/components/SidebarSearchModal/SidebarSearchModal.tsx#L37-L64

Reply:
We are using:
```
<SidebarPage>
    <Sidebar>
      <SidebarLogo />
      <SidebarGroup label="Search" icon={<SearchIcon />} to="/search">
        <SidebarSearchModal>
          {({ toggleModal }) => <SearchModal toggleModal={toggleModal} />}
        </SidebarSearchModal>
```
We haven't done anything custom to `SearchModal.tsx` so that's just using:
```
    "@backstage/plugin-search": "^1.0.6",
    "@backstage/plugin-search-react": "^1.3.1",
```

Reply:
![backstagesearch](https://user-images.githubusercontent.com/11878464/215563264-faf86e87-5c33-48d2-b5f8-7ab024fd2d34.gif)

Reply:
Correct, the modal is not closing when I click outside. I can add a close button (which is what I was originally thinking but even just fixing the behavior of clicking outside would be sufficient)

Reply:
Thank you for responding so quickly. I have a follow-up question:
Are you importing both `SidebatSearchModal` and `SearchModal` components from `@backstage/plugin-search` package, right? Could you bump the packages to the latest version and test closing the modal using the backdrop again? I couldn't simulate the reported behavior on the latest versions...

Latest versions:
@backstage/plugin-search -> 1.0.8-next.0
@backstage/plugin-search-react -> 1.4.0

Reply:
I managed to simulate here, let me debug and see if I can figure out what's happening:https://github.com/camilaibs/backstage-app-example/blob/master/packages/app/src/components/Root/Root.tsx#L64

Reply:
I think I get it; it looks like you're trying to render the `SearchModal` manually, ending up with two overlapping dialogs.
The first is [automatically added](https://github.com/backstage/backstage/blob/master/plugins/search/src/components/SidebarSearchModal/SidebarSearchModal.tsx#L50) by `SidebarSearchModal`, and the second by you when rendering a `SearchModal` again as children.

> ```jsx
> <SidebarPage>
>     <Sidebar>
>       <SidebarLogo />
>       <SidebarGroup label="Search" icon={<SearchIcon />} to="/search">
>         <SidebarSearchModal>
>           {({ toggleModal }) => <SearchModal toggleModal={toggleModal} />}
>         </SidebarSearchModal>
> ```

Due to this, when you click on the background, it just closes the parent modal but maintains the content you passed as children on the screen, which is the second `SearchModal`. The second modal is still on screen because you do not pass an open prop to it, and this prop is by default always "true" [here](https://github.com/backstage/backstage/blob/master/plugins/search/src/components/SearchModal/SearchModal.tsx#L172) in this line (you are passing only a `toggleModal` function). I created an example for you to validate what I'm saying, providing the open state by hand in [this](https://github.com/camilaibs/backstage-app-example/blob/588b6787d4a21600f9d3d1649f437431ef83aa74/packages/app/src/components/Root/Root.tsx#L64) file.

In light of that, my guidance is as follows:

If you want to use the default modal along with the sidebar, don't pass children like this:
```jsx
<SidebarGroup label="Search" icon={<SearchIcon />} to="/search"> 
  <SidebarSearchModal />
</SidebarGroup>
```

In this case, the `SidebarSearchModal` will render the `SearchModal` providing an open state to it, see [here](https://github.com/backstage/backstage/blob/master/plugins/search/src/components/SidebarSearchModal/SidebarSearchModal.tsx#L50).

Or, to customize the content of the search modal, you can pass your content component directly without wrapping it in a search modal:
```jsx
<SidebarGroup label="Search" icon={<SearchIcon />} to="/search"> 
  <SidebarSearchModal>
    {({ toggleModal }) => <CustomModalContent toggleModal={toggleModal} />}
  </ SidebarSearchModal>
</SidebarGroup>
```

Let me know if this comment clarify what is happening 🙂

Reply:
Sorry I'm getting to all of these comments out of order, so I just created a new issue.  But let me try this now and see if it resolves and I'll close the PR and the new issue (doh!)

Reply:
Okay, that worked!  Thank you so much!
I'll close this PR.

Reply:
Looking forward to seeing you in other contributions 🙌🏻